AFQ/data.py

```diff
@@ -862,13 +864,21 @@ class that quacks like AFQ.data.S3BIDSSubject. Default:
         self._non_subject_s3_keys = self._get_non_subject_s3_keys()
 
         # Convert `subjects` into a sequence of subjectID strings
-        if subjects is None or isinstance(subjects, int):
+        if subjects is None or isinstance(subjects, int) \
+                or (isinstance(subjects, list)
+                    and isinstance(subjects[0], int)):
             # if subjects is an int, get that many random subjects
-            n_subs = subjects if subjects is not None else 1
             prng = np.random.RandomState(random_seed)
-            subjects = prng.choice(sorted(self._all_subjects),
-                                   size=subjects,
-                                   replace=False)
+            randomized_subjects = self._all_subjects.copy()
```

Comment:
```suggestion
            randomized_subjects = sorted(self._all_subjects.copy())
```

Suggest sorting this since I'm not positive that the matching S3 keys will always be returned in the same order.

Reply:
Ah I see. good catch!